### PR TITLE
Add payment callback

### PR DESCRIPTION
**Pull request by PinkCode15**

### CODE REVIEW

File: app/PaymentProviders/Paystack.php [hunk 1]
```diff
@@ -42,7 +42,8 @@ public function initiatePayment(array $data): array
             'reference' => $data['reference'],
             'metadata' => [
                 'pricing_plan_uid' =>  $data['plan']->uid,
-            ]
+            ],
+            'callback_url' => config('app.url') . "/api/payment/callback?planuid={$data['plan']->uid}&email={$data['email']}"
```
Comment (IDTitanium) [inline app/PaymentProviders/Paystack.php:46]: Callback URL should be a frontend URL. It's the frontend that will redirect to the URL after successful payment.

File: routes/api.php [hunk 2]
```diff
@@ -72,6 +72,7 @@
 
 Route::group(['prefix' => 'payment'], function () {
     Route::post('', [PlanPaymentController::class, 'initiatePayment'])->name('payment');
+    Route::get('callback', [PlanPaymentController::class, 'processCallback'])->name('payment-callback');
```
Comment (IDTitanium) [inline routes/api.php:75]: We don't need this.

File: app/Actions/PlanPayment/ProcessCallbackAction.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    */
+    public function execute(Request $request): RedirectResponse
+    {
+        $pricingPlanUid = $request->planUid;
+        $email = $request->email;
+        $reference = $request->reference;
+
+        $url = config('app.website_url') ."/payment/callback?planuid=$pricingPlanUid&email=$email&reference=$reference";
+
+        return redirect()->to($url);
```
Comment (IDTitanium) [inline app/Actions/PlanPayment/ProcessCallbackAction.php:26]: Redirect happens on the website. Paystack will call the URL once payment is done.

File: app/PaymentProviders/Paystack.php (hunk 1 above)
Comment (PinkCode15) [reply app/PaymentProviders/Paystack.php:46]: Oh okay looks like I didn't really get what you trying to say. Updating in a bit

**Review by IDTitanium — APPROVED** [review]
The frontend needs to be aware of that route. And implement it.

**Pull request merged**